docs/src/content/docs/book/statements.mdx

```diff
@@ -96,6 +98,131 @@ value += 5;     // augmented assignment (one of the many, see below)
+    let Example { number } = Example { number: 42 };
+    //  -------   ------     ----------------------
+    //  ↑         ↑          ↑
+    //  |         |          instantiation of Example Struct
+    //  |         definition of "number" variable, derived
+    //  |         from the field "number" on Example Struct
+    //  target structure type "Example"
+    //  to destructure fields from
+
+    // The syntax above is equivalent to the following series of statements:
+    let example = Example { number: 42 };
+    let number = example.number;
+}
+```
+
+Just like in [instantiation](/book/expressions#instantiation), the trailing comma is allowed.
+
+```tact
+struct Example { number: Int }
+
+fun trailblazing() {
+    let Example {
+        number,     // trailing comma inside variable list
+    } = Example {
+        number: 42, // trailing comma inside field list
+    };
+}
+```
+
+:::note
+
+  [Augmented assignment operators](/book/operators#augmented-assignment) do not make sense for such assignments and will therefore be reported as parsing errors:
+
+  ```tact
+  struct Example { number: Int }
+
+  fun basic() {
+      let Example { number } += Example { number: 42 };
+      //                     ^ this will result in the parse error:
+      //                     expected "="
+  }
+  ```
+
+:::
+
+To create a binding under a different variable name, specify it after the semicolon `:{:tact}`.
+
+```tact
+// Similar definition, but this time field is called "field", not "number"
+struct Example { field: Int }
+
+fun naming(s: Example) {
+    let Example { field: varFromField } = s;
+    //                   ------------     ↑
+    //                   ↑                |
+    //                   |                instance of Example Struct, received
+    //                   |                as a parameter of the function "naming"
+    //                   definition of "varFromField" variable, derived
+    //                   from the field "field" on Example Struct
+}
+```
+
+Note, that the order of bindings doesn't matter — all the fields retain their values and types under their names no matter the order in which they stand in their definition in the respective [Struct][s] or [Message][m].
+
+```tact
+// "first" goes first, then goes "second"
+struct Two { first: Int; second: String }
+
+fun order(s: Two) {
+    let Two { second, first } = s;
+    //        ------  -----
+    //        ↑       ↑
+    //        |       this variable will be of type Int,
+    //        |       same as the "first" field on Struct Two
+    //        this variable will be of type String,
+    //        same as the "second" field on Struct Two
+}
+```
+
+Destructuring assignment is exhaustive and requires specifying all the fields as variables. To deliberately ignore some of the fields, use an underscore `_{:tact}`, which would make fields value considered unused and discarded. Note, that such wildcard variable name `_{:tact}` cannot be accessed:
```

Comment:
It's not exhaustive, you can say `let Two { first } = s` which is equivalent to `let Two { second: _, first } = s` (as in the example below)

Reply:
Curiosly, that depends. The following example:

```tact
struct Two { first: Int; second: String }

fun discard(s: Two) {
    let Two { first } = s;
    // ^^^^^^^^^^^^^^^^^^^
}
```

Will report an error: "Cannot evaluate expression to a constant: destructuring assignment expected 2 fields, but got 1".

But if one would place the same destructuring statement in the `receive()` function, there would be no such error:

```tact
struct Two { first: Int; second: String }
message HasTwo { s: Two }

contract {
   receive(msg: HasTwo) {
       // Almost the same line as in the previous example
       let Two { first } = msg.s;
       
       // No errors before or here
       dump(first);
   }
}
```

Reply:
Oh! I forgot to remove this error from the interpreter:
https://github.com/tact-lang/tact/blob/9522f818503acd253b0328d897b774134eca2548/src/interpreter.ts#L1468-L1475

I'll open a PR with a fix.

Reply:
Oh! I forgot to remove this error from interpreter. I'll open a fix PR:
https://github.com/tact-lang/tact/blob/9522f818503acd253b0328d897b774134eca2548/src/interpreter.ts#L1468-L1475